Mathlib/Data/Polynomial/Basic.lean

```diff
@@ -1221,10 +1221,10 @@ section repr
 variable [Semiring R]
 
 protected instance repr [Repr R] [DecidableEq R] : Repr R[X] :=
-  ⟨fun p prec =>
+  ⟨fun p _ =>
     if p.support = ∅ then "0"
     else
-      Repr.addAppParen
+      Lean.Format.paren
```

Comment:
This looks wrong to me, I think the old version was correct?

Reply:
I've reverted this and added a test to verify the original was correct.

Reply:
@eric-wieser `Repr.addAppParen` will add parentheses when the polynomial is being printed in a function argument context, but what about when you're printing it in place of `p` in, for example, `p * 2`? Even though `Lean.Format.paren` adds some superfluous parentheses, at least it correctly adds parentheses in this example.

Reply:
What do you mean by "printing in place of `p`"? Can you add a test for that?

Reply:
> It doesn't seem worth thinking too hard about being efficient about parentheses, unless you imagine this Repr instance is actually being used.

I think it's worth it so that we have an example of how `reprPrec` is actually intended to be used. I've pushed a fix and some tests, thanks for the explanation!